src/main/kotlin/no/kartverket/matrikkel/bygning/models/BygningExtensions.kt

```diff
@@ -7,105 +7,97 @@ package no.kartverket.matrikkel.bygning.models
 
 fun Bygning.withEgenregistrertData(egenregistreringer: List<Egenregistrering>): Bygning {
     return egenregistreringer.fold(this) { bygningAggregate, egenregistrering ->
+        val metadata = RegisterMetadata(
+            registreringstidspunkt = egenregistrering.registreringstidspunkt,
+            registrertAv = egenregistrering.eier.getValue(),
+        )
+
         bygningAggregate.copy(
             byggeaar = bygningAggregate.byggeaar.aggregate {
                 egenregistrering.bygningRegistrering.byggeaarRegistrering?.let {
                     Byggeaar(
                         data = it.byggeaar,
-                        metadata = RegisterMetadata(
-                            registreringstidspunkt = egenregistrering.registreringstidspunkt,
-                        ),
+                        metadata = metadata,
                     )
                 }
             },
             bruksareal = bygningAggregate.bruksareal.aggregate {
                 egenregistrering.bygningRegistrering.bruksarealRegistrering?.let {
                     Bruksareal(
                         data = it.bruksareal,
-                        metadata = RegisterMetadata(
-                            registreringstidspunkt = egenregistrering.registreringstidspunkt,
-                        ),
+                        metadata = metadata,
                     )
                 }
             },
             vannforsyning = bygningAggregate.vannforsyning.aggregate {
                 egenregistrering.bygningRegistrering.vannforsyningRegistrering?.let {
                     Vannforsyning(
                         data = it.vannforsyning,
-                        metadata = RegisterMetadata(
-                            registreringstidspunkt = egenregistrering.registreringstidspunkt,
-                        ),
+                        metadata = metadata,
                     )
                 }
             },
             avlop = bygningAggregate.avlop.aggregate {
                 egenregistrering.bygningRegistrering.avlopRegistrering?.let {
                     Avlop(
                         data = it.avlop,
-                        metadata = RegisterMetadata(
-                            registreringstidspunkt = egenregistrering.registreringstidspunkt,
-                        ),
+                        metadata = metadata,
                     )
                 }
             },
+            bruksenheter = bygningAggregate.bruksenheter.map {
+                it.applyEgenregistrering(egenregistrering)
+            },
         )
     }
 }
 
+private fun Bruksenhet.applyEgenregistrering(egenregistrering: Egenregistrering): Bruksenhet {
+    val bruksenhetRegistrering =
+        egenregistrering.bygningRegistrering.bruksenhetRegistreringer.firstOrNull { it.bruksenhetId == this.bruksenhetId }
+    if (bruksenhetRegistrering == null) return this
```

Comment:
Litt pirk og personlig smak kanskje, men tenker generelt det er fint med `{` selv på one linere.
```suggestion
    if (bruksenhetRegistrering == null) {
    return this
}
```